arelle/model/ModelTransition.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+from typing import Any
+
+from lxml import etree
+
+from arelle.model.ModelCommentBase import ModelCommentBase
+from arelle.model.ModelElementBase import ModelElementBase
+from arelle.model.ModelPIBase import ModelPIBase
+
+_LXML_TO_NEW_BASE_CLASS_MODEL_MAPPING = {
+    etree.CommentBase: ModelCommentBase,
+    etree.ElementBase: ModelElementBase,
+    etree.PIBase: ModelPIBase,
+}
+
+_METACLASS_USED = False
+_USE_NON_LXML_BASE_CLASSES = False
+
+
+def useNewModelBaseClasses() -> None:
+    global _METACLASS_USED, _USE_NON_LXML_BASE_CLASSES
+    if _USE_NON_LXML_BASE_CLASSES:
+        return
+    if _METACLASS_USED:
+        raise RuntimeError("LXML metaclass model transition must be set before any model classes are defined")
+    _USE_NON_LXML_BASE_CLASSES = True
+
+
+class LXMLModelTransitionMetaclass(type):
+    def __new__(cls, clsname: str, bases: tuple[type, ...], attrs: dict[str, Any]) -> type:
+        global _METACLASS_USED
+        _METACLASS_USED = True
+        if _USE_NON_LXML_BASE_CLASSES:
```

Comment:
Out of curiosity, why isn't `global` needed for `_USE_NON_LXML_BASE_CLASSES` or `_LXML_TO_NEW_BASE_CLASS_MODEL_MAPPING `?

Reply:
Global is only required if the variable is assigned to. Thanks for pointing this out. I had a global `_METACLASS_USED` in `useNewModelBaseClasses` that was unneeded

Reply:
Ah, got it, thanks.  You correctly deduced I was comparing to `useNewModelBaseClasses` :-).